src/web/app/pages-instructor/instructor-student-list-page/instructor-student-list-page.component.ts

```diff
@@ -12,13 +69,494 @@ import { ActivatedRoute } from '@angular/router';
+      }
+    });
+  }
+
+  /**
+   * Function to change states of all courses.
+   */
+  toggleAllPresentCoursesStateAtInput(defaultState: boolean): void {
+    this.allPresentCourses.forEach((course: Course) => {
+      if (course) {
+        this.toggleCourseStateAtInput(course, defaultState);
+      }
+    });
+  }
+
+  /**
+   * Change the state of all sections presented on the screen.
+   */
+  toggleAllPresentSectionDetailsState(defaultState: boolean): void {
+    this.allPresentSectionDetails.forEach((sectionDetails: SectionDetails) => {
+      this.toggleSectionState(sectionDetails, defaultState);
+    });
+  }
+
+  /**
+   * Change the state of all teams presented on the screen.
+   */
+  toggleAllPresentTeamDetailsState(defaultState: boolean): void {
+    this.allPresentTeamDetails.forEach((teamDetails: TeamDetails) => {
+      this.toggleTeamState(teamDetails, defaultState);
+    });
+  }
+
+  /**
+   * Change the state of all students presented on the screen.
+   */
+  toggleAllPresentStudentDetailsState(defaultState: boolean): void {
+    this.allPresentStudentDetails.forEach((studentDetails: StudentDetails) => {
+      this.toggleStudentState(studentDetails, defaultState);
+    });
+  }
+
+  /**
+   * Function to trigger getting CourseDetails data for a course from backend.
+   */
+  toggleCourseStateAtInput(course: Course, designatedState?: boolean): void {
+    if (course.isChecked === undefined) {
+      course.isChecked = false;
+    }
+    const state: boolean = designatedState === undefined
+        ? !course.isChecked
+        : designatedState;
+
+    this.fetchCourseDetails(course.id)
+        .subscribe((courseDetails: CourseDetails | null) => {
+          if (courseDetails) {
+            this.toggleCourseState(courseDetails, state);
+            this.courseStudentListSectionDataMap[courseDetails.id]
+                = this.getStudentListSectionDataFromCourseDetails(courseDetails);
+          }
+        }, (error: any) => {
+          this.statusMessageService.showErrorMessage(error);
+        });
+  }
+
+  /**
+   * Function to call backend API to get CourseDetails data for a course.
+   */
+  fetchCourseDetails(courseid: string): Observable<CourseDetails | null> {
+    if (this.isCourseDetailsInList(courseid)) {
+      return of(this.getCourseDetails(courseid));
+    }
+
+    const paramMap: { [courseid: string ]: string } = { courseid };
+    return this.httpRequestService.get('/instructor/students', paramMap)
+        .pipe(map((resp: GetCourseDetailsResponse) => {
+          if (resp.course) {
+            resp.course.isChecked = false;
+            this.courseDetailsList.push(resp.course);
+            return this.getCourseDetails(resp.course.id);
+          }
+
+          this.statusMessageService.showErrorMessage(`Error retrieving course details for course of id ${courseid}`);
+          return null;
+        }));
+  }
+
+  /**
+   * If the courseDetails of a course is fetched.
```

Comment:
- You seem to miss `Returns true` at the beginning.
- "is fetched" and "in list" doesn't seem to tally.